### DISCUSSION TITLE
do packages that use ts-toolbelt a major upgrade for ts-toolbelt v9?

### DISCUSSION THREAD

**Discussion by mesqueeb** [millsp/ts-toolbelt#200] [Q&A] [answered]
## 🤔 Question

2 short questions!

1. if my "util" package has `ts-toolbelt` as dependency. Should I make TS-Toolbelt a `dependency` a `devDependency` or a `peerDependency` ?
2. if my package upgrades to ts-toolbelt v9, should it do a major upgrade?

**Comment by millsp** [comment, marked as answer]
1. It must be a `dependency`. If you set it as a `devDependency`, your package consumers will not benefit from the types packaged in the `ts-toolbelt` - it will appear to be broken at first. They might have to install `ts-toolbelt` by hand, that's why you could enable `peerDependency`. But since two `ts-toobelt` versions can now cohabitate, it really only makes sense to add it as a direct `dependency`.

2. The short answer is "No". The types are safer, so stuff might break. But we can generally consider this as a patch - not a breaking change. These changes mostly affect output of types that use the `At` utility, because `At` will now yield the `undefined` type when props don't exist  (from version `9` on) - whereas it used to yield `never` and that could cause problems on unions, arrays, etc... The breaking change is more for you, who will have to do the work to adapt to the new API.
